src/utils/appLinking.js

```diff
@@ -0,0 +1,125 @@
+		((intent === "channel" && (inquiry = inquiry in slackInfo.channel ? slackInfo.channel[inquiry] : inquiry))
+			? { uri: `slack://channel?team=${slackInfo.team}&id=${inquiry}` }
+			: { uri: `slack://open?team=${slackInfo.team}` }),
+	phone: (inquiry) =>
+		({ uri: `tel:+${inquiry}`, url: `tel:+${inquiry}` }),
+	web: (inquiry) =>
+		({ uri: `${inquiry}`, url: `${inquiry}` })
+};
+
+/**
+ * @description This function handles the action launching an app, like a mail app, from our app.
+ *              It uses uri schemes and universal links to do deep linking in both platforms.
+ *
+ *  ** It requires three parameters and an optional warning message. **
+ *  With Slack, write the name of the channel or use slackInfo to see the channels available.
+ *
+ * @param {("email"|"facebook"
+ *         |"instagram"|"linkedin"
+ *         |"slack"|"phone"|"web")} appName         The app name to be open.
+ * @param {("groups"|"profile"
+ * 		   |"company"|"channel"
+ * 		   |"open")} 				intent          The intent of the search.
+ * @param {("announcements"
+ * 		   |"fundraisingCommitte"
+ * 		   |"general"|"motorshpe"
+ * 		   |"projects"|"shpejr")}   inquiry   	    The value for the query search. Slack channels are shown here for easiness.
+ * @param {String=} 				warningMessage  Optional value for the alert. It use to populate the alert dialog box.
+ *
+ *  -- How to use the function --
+ *  Calling the function passing 4 params, the name of the app, the action, what you are looking and the
+ *  alert message.(appName, intent, inquiry, Warningmessage)
+ *
+ *  Calling the function passing 3 param, the app name, the action, and what you are looking.(appName, intent, inquiry)
+ *
+ * @example
+ * // slack: To open the chapter worspace
+ * openAppOrWebsite({"slack"}, {"open"}, {"team" | slackInfo.team}, "Optional error  msg");
+ * // slack: To send the user directly to a channel, let's say general
+ * openAppOrWebsite({"slack"}, {"channel"}, {"general" | slackInfo.channel.general}, "Optional error  msg");
+ *
+ * // linkedin (profile or comapny): To open a profile page.
+ * openAppOrWebsite({"linkedin"}, {"profile"}, {activeUser.linkedin | "companyProfile"}, " something ...")
+ *
+ * // instagram (profile), facebook (profile or group): To open a profile or group page
+ * openAppOrWebsite({"appName"}, {"profile" | "group"}, {userProfile}, " something ...")
+ *
+ * // email, phone or website:
+ * openAppOrWebsite({"mail" | "phone"}, {"open"}, {activeUser.email | "14075551010"}, " something ...")
+ * openAppOrWebsite("web", "open", {"https://www.example.com/"}, " something ...")
+ */
+
+export function openAppOrWebsite(appName, intent, inquiry, warning) {
+	warning = (warning) ? warning : "Oops! Something went wrong.\nContact Tech Director if problem persist.";
+
+	if (!inquiry) {
+		const name = appName[0].toUpperCase() + appName.slice(1);
+
+		Alert.alert(warning, { title: `Missing ${name} information.`, type: "alert", submit: { title: "Close" } });
```

Comment:
Sorry about before, the default `type` is "alert" so there's no need to include it